src/Ionide.ProjInfo/Types.fs

```diff
@@ -38,9 +39,18 @@ module Types =
     type Property = { Name: string; Value: string }
 
     type PackageReference = {
+        /// Name of the NuGet package, e.g. "FSharp.Core".
         Name: string
+        /// Version of the NuGet package, e.g. "8.0.400".
         Version: string
+        /// Path to the DLL on disk whose import is implied by this PackageReference.
+        /// This is likely somewhere within the local NuGet cache.
+        ///
+        /// E.g. "/Users/user/.nuget/packages/fsharp.core/6.0.1/lib/netstandard2.1/FSharp.Core.dll".
         FullPath: string
+        /// See e.g. https://github.com/dotnet/sdk/blob/6ef345b1ff76eec2de6a0deeda321cae45da655b/src/Tasks/Common/MetadataKeys.cs#L65
+        /// for a large set of keys which may or may not be present in this metadata dict.
```

Comment:
(This all seems to be completely undocumented and frankly quite terrifying.)

Reply:
It's not undocumented, it's just that MSBuild is an open system and any single piece of MSBuild logic can add or interpret metadata on MSBuild Items however it wishes. 

As a result, certain patterns and well known metadata keys (ideally documented) become the contracts. 

For example, here's the metadata that MSbuild gives you for any Item: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2022

Reply:
Generally, if a specific piece of metadata is required, e.g. TargetPath, you'd make a Task who's job it is to take Items as input and add TargetPath metadata to all of the items. Then you'd put that Task in a Target so that folks have a publicly-visible way to ensure that a given group of Items gets that metadata.

Reply:
Thanks; sounds like it's finally time for me to read the MSBuild docs end-to-end :( 

In that case I will not attempt to document the metadata field.